### PR TITLE
Add service-specific timeout configuration for builderclient interactions

### PR DESCRIPTION
**Pull request by hoanhan101**
## Description

This PR implements service-specific timeout configuration for builderclient interactions with MEV-boost relays. The current implementation uses a single global timeout for all builderclient operations, but different services (validator registration, bid fetching, block unblinding) have different performance characteristics and would benefit from tailored timeout values.

The solution adds hierarchical timeout configuration following the pattern `builderclient.{service}.{address}.timeout` with service-specific defaults: 10s for blockrelay operations, 5s for validator registrations and builder bid strategies.

## Changes

- Add service-specific timeout configuration pattern `builderclient.{service}.{address}.timeout`
- Implement default timeouts: 10s for blockrelay, 5s for submitvalidatorregistrations and strategies.builderbid
- Add hierarchical timeout resolution with fallback chain: service+address → service → builderclient → global
- Update builderclient caching to use service-specific cache keys
- Add comprehensive test coverage for timeout resolution and edge cases
- Update configuration documentation with builderclient timeout examples
- Optimize test suite by consolidating helper functions and removing code duplication
- Consolidate duplicate path construction logic in FetchBuilderClient

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -246,6 +248,26 @@ blockrelay:
...
       factor: 0
 
+# builderclient provides timeout configuration for builder/relay communication.
+# Vouch communicates with MEV-boost relays for validator registration, bid fetching, and block unblinding.
+# Different operations have different performance characteristics and may benefit from different timeout values.
+builderclient:
+  # Global timeout for all builderclient operations. Falls back to the global 'timeout' value if not set.
+  timeout: '10s'
+
+  # Validator registration timeout (default: 5s)
```
Comment (Bez625) [inline docs/configuration.md:258]: By convention we set the values in the config to be the defaults. We can remove the comment `(default: 5s)` in this case. We should probably call out that this defaults to builderclient.timeout if not set.

File: docs/configuration.md [hunk 2]
```diff
@@ -246,6 +248,26 @@ blockrelay:
       # With a factor of 0 bids from this builder will be ignored.
       factor: 0
 
+# builderclient provides timeout configuration for builder/relay communication.
+# Vouch communicates with MEV-boost relays for validator registration, bid fetching, and block unblinding.
+# Different operations have different performance characteristics and may benefit from different timeout values.
+builderclient:
+  # Global timeout for all builderclient operations. Falls back to the global 'timeout' value if not set.
+  timeout: '10s'
```
Comment (Bez625) [inline docs/configuration.md:256]: Is the default value really 2s if this isn't set? If so we should change the value to that.

File: docs/configuration.md [hunk 3]
```diff
@@ -246,6 +248,26 @@ blockrelay:
...
+
+  # Validator registration timeout (default: 5s)
+  submitvalidatorregistrations:
+    timeout: '30s'
```
Comment (Bez625) [inline docs/configuration.md:260]: The default here should be `10s`

File: docs/configuration.md [hunk 4]
```diff
@@ -246,6 +248,26 @@ blockrelay:
...
+
+  # Builder bid strategy timeout (default: 5s)
+  strategies:
```
Comment (Bez625) [inline docs/configuration.md:263]: How come we need strategies here?

File: docs/configuration.md [hunk 5]
```diff
@@ -246,6 +248,26 @@ blockrelay:
...
+    builderbid:
+      timeout: '2s'
```
Comment (Bez625) [inline docs/configuration.md:265]: The default here should be `5s`

File: docs/configuration.md [hunk 6]
```diff
@@ -246,6 +248,26 @@ blockrelay:
...
+
+  # Block relay timeout for unblinding operations (default: 10s)
+  blockrelay:
+    timeout: '15s'
```
Comment (Bez625) [inline docs/configuration.md:269]: The default here should be `5s`

**Review by Bez625 — CHANGES_REQUESTED** [review]

File: docs/configuration.md [hunk 7]
```diff
@@ -304,62 +328,86 @@ submitter:
...
+
+- `builderclient.strategies.builderbid.relay1.flashbots.net.timeout`
+- `builderclient.strategies.builderbid.timeout`
+- `builderclient.timeout`
+- `timeout`
+
+If none of these are explicitly configured, service-specific defaults are applied:
+
+- `blockrelay`: 5s default
+- `submitvalidatorregistrations`: 5s default
```
Comment (Bez625) [inline docs/configuration.md:352]:
```suggestion
- `submitvalidatorregistrations`: 10s default
```

File: docs/configuration.md [hunk 8]
```diff
@@ -246,6 +248,27 @@ blockrelay:
...
+# builderclient provides timeout configuration for builder/relay communication.
+# Vouch communicates with MEV-boost relays for validator registration, bid fetching, and block unblinding.
+# Different operations have different performance characteristics and may benefit from different timeout values.
+builderclient:
+  # Global timeout for all builderclient operations. Falls back to the global 'timeout' value (2s) if not set.
+  timeout: '2s'
+
+  # Validator registration timeout
+  submitvalidatorregistrations:
+    timeout: '5s'
```
Comment (Bez625) [inline docs/configuration.md:260]:
```suggestion
    timeout: '10s'
```

File: util/builders.go [hunk 9]
```diff
@@ -26,41 +26,70 @@ import (
 	"github.com/spf13/viper"
 )
 
+// SetServiceDefaults sets default timeouts for builderclient services only if not already configured.
+func SetServiceDefaults() {
+	defaults := map[string]string{
+		"builderclient.blockrelay.timeout":                   "5s",
+		"builderclient.submitvalidatorregistrations.timeout": "5s",
```
Comment (Bez625) [inline util/builders.go:33]:
```suggestion
		"builderclient.submitvalidatorregistrations.timeout": "10s",
```

**Review by Bez625 — CHANGES_REQUESTED** [review]
Lets make the default for validator registrations 10s and then we are good to go.
